src/runtime/threads/topology.cpp

```diff
@@ -678,7 +707,7 @@ namespace hpx { namespace threads
 
     std::size_t topology::get_number_of_numa_nodes() const
     {
-        int nobjs =  hwloc_get_nbobjs_by_type(topo, HWLOC_OBJ_NODE);
+        int nobjs =  hwloc_get_nbobjs_by_type(topo, HWLOC_OBJ_NUMANODE);
```

Comment:
Will `HWLOC_OBJ_NUMANODE` be available for older versions of hwloc?

Reply:
I've just removed that commit. The reason is that when `HWLOC_OBJ_NUMANODE` does exist it's an enum value, and thus we can't check `if defined(HWLOC_OBJ_NUMANODE)`. The version check that we had should be sufficient and what is recommended by hwloc for compatibility.